website/docs/opengl/c1/2-hello-quad.md

```diff
@@ -0,0 +1,524 @@
+
+#### Creating the fragment shader
+Our fragment shader for this example is very simple.
+
+Add the following to your `OnLoad` method:
+
+```cs
+const string fragmentCode = @"
+#version 330 core
+
+out vec4 out_color;
+
+void main()
+{
+    out_color = vec4(1.0, 0.5, 0.2, 1.0);
+}";
+```
+
+In our fragment shader, we simply output a single color, in this case a reddish-orange.
+
+Because our fragment shader is so simple, we don't use any `in` attributes for this example. However, unlike a vertex shader, a fragment shader must **always** have at least one `out` attribute. This attribute is the output color of the fragment shader itself, and must always be assigned a value. (We'll get more into this in a later tutorial). The output color is in the RGBA format. In OpenGL, the output color in the fragment shader is a **normalized** 32-bit float. Therefore, each of the RGBA values must be between 0 and 1. This is also true for our clear color too, however Silk.NET handily accepts a `System.Drawing.Color`, which we are using in this tutorial.
+
+#### Creating the shader objects
+Let's create the shader objects!
+
+Add the following to your `OnLoad` method:
+
+```cs
+uint vertexShader = _gl.CreateShader(ShaderType.VertexShader);
+_gl.ShaderSource(vertexShader, vertexCode);
+```
+
+First, we create the shader object itself. Each shader object must be assigned a **type**, in this case the vertex shader. Then we simply give the shader the code we defined earlier.
+
+Now we need to **compile** the shader.
+
+Add the following to your `OnLoad` method:
+
+```cs
+_gl.CompileShader(vertexShader);
+
+_gl.GetShader(vertexShader, ShaderParameterName.CompileStatus, out int status);
+if (status != (int) GLEnum.True)
+    throw new Exception("Vertex shader failed to compile: " + _gl.GetShaderInfoLog(vertexShader));
+```
+
+First, we compile the shader. Then, we need to check to make sure it has compiled correctly, and if it hasn't, it needs to let us know what our error is. `glGetShader` can give us a lot of different parameters, some of which we'll be using in later tutorials. For now, we are just interested in the compile status.
+
+The compile status returns `1` if successful, and `0` if not. If it's not successful, we throw an exception. `glGetShaderInfoLog` gets the error string for the shader, and we output that in our exception. You don't *have* to throw an exception though, you can do whatever you wish with this result. Just don't try to continue further, as you will get a link error! (We'll get to this in a minute).
+
+We've now successfully created and compiled our vertex shader. To compile our fragment shader, simply copy and paste the code you just wrote. Just make sure to change the `ShaderType` to `FragmentShader`!
+
```

Comment:
```suggestion
We've now successfully created and compiled our vertex shader. To compile our fragment shader, simply copy and paste the code you just wrote. Just make sure to change the `ShaderType` to `FragmentShader`!

```cs
uint fragmentShader = _gl.CreateShader(ShaderType.FragmentShader);
_gl.ShaderSource(fragmentShader, fragmentCode);
_gl.CompileShader(fragmentShader);

_gl.GetShader(fragmentShader, ShaderParameterName.CompileStatus, out int status);
if (status != (int) GLEnum.True)
    throw new Exception("Fragment shader failed to compile: " + _gl.GetShaderInfoLog(fragmentShader));
```

```

Reply:
I decided not to include the fragment shader code as it *should* be fairly obvious to the reader what needs to be copy-pasted and changed (as I even prompt them to do so and what to change), so I'd rather not include the duplicate code, unless you really think I should. If they get really stuck they could just look at the final result code.

Reply:
I think you should include it, mostly for those that are just copy-pasting, but also to show where the `fragmentShader` variable comes from.

Reply:
Alrighty, I'll include it in the tutorial